cranelift-wasm/src/environ/spec.rs

```diff
@@ -144,6 +144,12 @@ pub trait FuncEnvironment: TargetEnvironment {
         signature.params[index].purpose == ir::ArgumentPurpose::Normal
     }
 
+    /// Is the given return of the given function a wasm-level parameter, as
+    /// opposed to a hidden parameter added for use by the implementation?
+    fn is_wasm_return(&self, signature: &ir::Signature, index: usize) -> bool {
+        signature.returns[index].purpose == ir::ArgumentPurpose::Normal
+    }
```

Comment:
> It's is_wasm_parameter because it isn't counting the number of ArgumentPurpose::Normal parameters. 

@sunfishcode, so the above is a default implementation that embedders are supposed to override?